spec/vmpooler/api/v1_spec.rb

```diff
@@ -265,6 +265,32 @@ def app()
         expect_json(ok = true, http = 200)
       end
 
+      it 'fails when not all requested vms can be allocated' do
+        allow(redis).to receive(:spop).with('vmpooler__ready__pool1').and_return 'abcdefghijklmnop'
```

Comment:
Don't really like these very mockist tests, but this is the shortest path from the current suite.

If we have a redis available, seems like we should set up state, perform operations, validate state, and then teardown state.